File: tang-commons/src/main/java/com/tang/commons/utils/page/PageUtils.java
package com.tang.commons.utils.page;

import java.util.List;

import com.github.pagehelper.PageHelper;
import com.github.pagehelper.PageInfo;
import com.github.pagehelper.page.PageMethod;
import com.tang.commons.constants.HttpStatus;
import com.tang.commons.utils.SqlUtils;

/**
 * 分页工具类
 *
 * @author Tang
 */
public class PageUtils extends PageHelper {

    private PageUtils() {
    }

    /**
     * 设置请求分页数据
     */
    public static void startPage() {
        var pageDomain = PageSupport.buildPageRequest();
        var pageNum = pageDomain.getPageNum();
        var pageSize = pageDomain.getPageSize();
        var orderBy = SqlUtils.escapeOrderBySql(pageDomain.getOrderBy());
        var reasonable = pageDomain.getReasonable();
        PageMethod.startPage(pageNum, pageSize, orderBy).setReasonable(reasonable);
    }

    /**
     * 清理分页的线程变量
     */
    public static void clearPage() {
        PageMethod.clearPage();
    }

    /**
     * 响应请求分页数据
     */
    public static PageResult getDataTable(List<?> list) {
        var tableDataResult = new PageResult();
        tableDataResult.setCode(HttpStatus.SUCCESS);
        tableDataResult.setMsg("查询成功");
        tableDataResult.setRows(list);
        tableDataResult.setTotal(new PageInfo<>(list).getTotal());
        return tableDataResult;
    }

}

File: tang-extension/tang-mybatis-flex/src/main/kotlin/com/tang/extension/mybatisflex/utils/PageUtils.kt
package com.tang.extension.mybatisflex.utils

import com.mybatisflex.core.BaseMapper
import com.mybatisflex.core.query.QueryWrapper
import com.tang.commons.utils.page.PageResult
import com.tang.commons.utils.page.PageSupport

/**
 * MyBatis Flex 分页工具类
 *
 * @author Tang
 */
object PageUtils {

    /**
     * 分页查询
     *
     * @param baseMapper   通用 Mapper 接口
     * @param queryWrapper 查询条件
     * @param <T>          实体类
     * @return 分页结果
     */
    @JvmStatic
    fun <T> page(baseMapper: BaseMapper<T>, queryWrapper: QueryWrapper): PageResult {
        val pageDomain = PageSupport.buildPageRequest()
        if (pageDomain.orderBy.isNotBlank()) {
            queryWrapper.orderBy(pageDomain.orderByColumn, pageDomain.isAscBoolean)
        }
        val paginate = baseMapper.paginate(pageDomain.pageNum, pageDomain.pageSize, queryWrapper)
        return PageResult(paginate.records, paginate.totalRow)
    }

}
